### DISCUSSION TITLE
Question : Why does rotating page doesn't change mediaBox dimensions ?

### DISCUSSION THREAD

**Discussion by yodapotatofly** [py-pdf/pypdf#704] [Q&A]
I have a document with some pages being landscape, and the others being portrait.

I rotate all landscape pages to portrait (by checking mediaBox parameters and comparing x and y)
Then, I want to crop the pages in order to save only the top of the pages. But when i tried to crop a page that I have rotated, then everything goes sideways.
Actualy, the mediaBox parameters doesn't change between before and after rotating, while I think it should invert x and y sizes ?

```
from PyPDF2 import PdfFileWriter, PdfFileReader


def rotateLandscapePages(pdf):
    pdf_out = PdfFileWriter()
    numPages=pdf.getNumPages()


    for pageIndex in range(numPages):
        page = pdf.getPage(pageIndex)


        widht= page.mediaBox[2]
        height= page.mediaBox[3]
        print("\nPage",pageIndex+1)
        print("----------")
        print("widht :",widht, "; height:", height)
        print(page.mediaBox)
        print(page.mediaBox.lowerRight)
        print(page.mediaBox.lowerLeft)
        print(page.mediaBox.upperRight)
        print(page.mediaBox.upperLeft)


        if largeur > hauteur:
            print("Format paysage                     ",end=" ")
            print("Rotation à gauche")
            newPage = page.rotateCounterClockwise(90)

        else:
            print("Format portrait                     ",end=" ")
            print("Pas de rotation")
            newPage = page

        widht = newPage.mediaBox[2]
        height= newPage.mediaBox[3]
        print("\nPage",pageIndex+1)
        print("----------")
        print("widht :",widht, "; height :", height)
        print(newPage.mediaBox)
        print(newPage.mediaBox.lowerRight)
        print(newPage.mediaBox.lowerLeft)
        print(newPage.mediaBox.upperRight)
        print(newPage.mediaBox.upperLeft)

        cropTopHalf(newPage)
        pdf_out.addPage(newPage)

    return pdf_out

def cropTopHalf(page):
    page.mediaBox.lowerLeft = ( page.mediaBox.getLowerLeft_y(), page.mediaBox.getUpperRight_y()/2.1 )

if __name__ == '__main__':
    path = 'landscape_pdf_file.pdf'

    pdf = PdfFileReader(path)

    pdf = rotateLandscapePages(pdf)

    with open('out.pdf', 'wb') as fh:
        pdf.write(fh)

```

**Comment by thommiano** [comment]
Similarly, I am rotating pages (using `page.RotateClockwise`) that have annotations, but I've observed that the mediaBox and the annotation Rects do not have updated coordinates that reflect that page rotation. The rotation attribute also does not get updated.

If I write a rotate page to disk, then read it back in, the page looks rotated to the eye, but the annotations still have their original coordinates.

The expected behavior is that mediaBox, CropBox, Annotations, etc. have coordinates that reflect the current orientation/rotation of the page.

**Comment by maimaik05** [comment]
I am having this same issue, hoping to bring this back up if there is any input!

**Comment by maimaik05** [comment]
I am having this same issue, hoping to bring this back up if there is any input!

**Comment by pubpub-zz** [comment]
@maimaik05 
many changes have to do to fix this.
a) can you confirm you are on pypdf 3.11
b) if you are still facing an issue, please fill up a issue with all requested data.

**Reply by maimaik05** [reply]
I just upgraded to 3.11 but nothing changed, my initial issue is that when i open the pdf with PdfReader, the pdf gets rotated from the original, so when i find the annotations on the pdf, is rotated 90 degrees counterclockwise from what I want. I can send in the code privately if needed.

**Reply by pubpub-zz** [reply]
Can you please clarify which function you are using : `PageObject.rotate()` is changing the `/Rotate` property which is asking the viewer to display the page rotated : the page (and the mediabox are unchanged)

**Reply by maimaik05** [reply]
I'm using PageObject.rotate() with PageObject.transfer_rotation_to_content() after that. Is there another function that can rotate the mediabox and the /Annots '/Rect' object?

**Reply by maimaik05** [reply]
The mediabox of the pdf after being opened with pypdf is [0, 0, 2592, 3456], which doesn't make sense because the original pdf that is being opened is landscape, so I think something is being rotated when it shouldn't be? That's why I am trying to figure out how to undo that rotation with the annotations, but if I can prevent the initial rotation that would be preferred.

**Reply by pubpub-zz** [reply]
can you publish your pdf for review

**Reply by maimaik05** [reply]
Unfortunately, I can't share it, but I can say that it's a floorplan with AutoCAD Text as comments in the pdf.

**Reply by maimaik05** [reply]
Originally we were taking the pdf and using Adobe Acrobat to export the pdf as a jpg, then using PyPDF to find the location of the comments (annotations), by finding the center of the rect box of the annotation. It was working fine with that process. We figured out how to convert from pdf to jpg using python, but now the location of the annotation boxes appear to be rotated.

**Reply by pubpub-zz** [reply]
there is no logic in the document production between actual page orientation and "/Rotate" field. What you can try is to call `PageObject.transfer_rotation_to_content()` : this will modify the actual pageMedia and integration rotation within the content : this may fix your issue

**Reply by mohit1337** [reply]
`PageObject.transfer_rotation_to_content()` fixed the issue

**Comment by mohit1337** [comment]
This is still a problem. In below screenshots you can see, first one is original pdf of dimension 792x612
<img width="284" alt="image" src="https://github.com/py-pdf/pypdf/assets/14182618/469eb146-3c79-45cc-a1c4-4781c3d455ec">

But even after rotating, the pdf is simply rotated to eye but dimensions are maintained

<img width="256" alt="image" src="https://github.com/py-pdf/pypdf/assets/14182618/3de8b94d-4b1c-49d5-ae2e-a8d9c1d38914">

**Reply by pubpub-zz** [reply]
@mohit1337 
can you please provide the PDF. if you prefer you can mail it privately to @MartinThoma

**Reply by stefan6419846** [reply]
The other comment (https://github.com/py-pdf/pypdf/discussions/704#discussioncomment-7172842) has been added three minutes later, so I assume this could be resolved as well.

**Reply by pubpub-zz** [reply]
Not understood
I would like to see the PDF to analyse the behavior

**Reply by mohit1337** [reply]
@pubpub-zz, this is not pdf dependant, you can select any pdf of a rectangular dimension, load it and call `.rotate(90)` on the page and save it again and you shall be able to reproduce this.
Calling `.transfer_rotation_to_content()` on the page just before writing it fixed the issue though. This should either be default behaviour or made explicit in documents.

**Reply by frankjalali** [reply]
I had the same issue. 
I have a CAD drawing, shown in Foxit in lanscape mode. But rotation was 270 and width and height were 22X34.
to change it a sound landscape mode, I tried all stuff from rotation to transformation. Nothing helped except for this .transfer_rotation_to_content().  I agree that this should be part of documentation. Here is the only place I could find it.

**Reply by pubpub-zz** [reply]
> I had the same issue. 
> I have a CAD drawing, shown in Foxit in lanscape mode. But rotation was 270 and width and height were 22X34.
> to change it a sound landscape mode, I tried all stuff from rotation to transformation. Nothing helped except for this .transfer_rotation_to_content().  I agree that this should be part of documentation. Here is the only place I could find it.

Do not hesitate to propose a PR to upgrade documentation

**Reply by stefan6419846** [reply]
There are at least two pages inside the documentation which already refer to the corresponding method (apart from the API docs).
